design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,239 @@
+This means that all environments must use the same version of the stack.
+If an admin wants to try a newer version of the stack for their dev environment, it will also result in the newer version of the stack being used in the production environment as well.
+
+This limitation could be addressed in the future by allowing multiple versions of cluster stacks installed per control plane, which would require additional complexity and design work.
+To get started, we will make the simplifying constraint that only a single version of a cluster stack can be installed and it will be shared across all namespaces (environments).
+
+#### Environment Policies
+
+Another potential advantage of supporting multiple namespace "environments" is that each one could be assigned their own specific policy, such as default resource classes.
+The production environment could have a completely different set of default resource classes than the development environment.
+Since these environments are simply represented by namespaces, they can take on any arbitrary meaning or scope.
+For example, instead of dev vs. prod, they could represent entirely other dimensions, such as a high performance environment, a cheap environment, an AWS only environment, etc.
+
+The full design for this support of different policies per environment is out of scope of this document, but this idea has interesting potential.
+Also note that in the current implementation of policies, they must be in the same namespace as the claim referencing them.
+This will likely be updated in the future to support cluster or environment scoped policies.
+
+## Namespace Stacks
+
+Namespace Stacks are likely to represent higher level "application" functionality, but are not required to do so.
+Instead of providing the infrastructure and lower level platform components for entire environments like a cluster stack, they likely will merely provide the functionality to support a single application.
+It is expected for namespace stacks to (indirectly) consume and build on functionality from cluster stacks in the form of resource claims.
+For example, a namespace stack could generate a `MySQLInstance` claim that is fulfilled through a default resource class by a cluster stack in the form of a concrete MySQL database instance.
+
+The key aspect of a namespace stack is that it will be installed with permissions to only watch for resources **within the namespace** it is running in.
+This provides a nice isolation as well as versioning story.
+
+* **Isolation:** The namespace stack should not be able to read or write any resources outside of its target namespace.
+* **Versioning:** Multiple versions of the namespace stack can be installed within the control plane, each namespace can have their own independent version.
+
+### Installation Flow
+
+Let's look at a concrete example of what happens when the `WordPress` stack is installed into Crossplane in a target namespace of the user's choice:
```

Comment:
It may be worth linking to the WordPress stack here if we are referring to it by name.

Reply:
I'll add a link to the WordPress stack repo